packages/dictionary/src/metaSchema/dictionarySchemas.ts

```diff
@@ -66,35 +67,112 @@ export type SchemaFieldValueType = zod.infer<typeof SchemaFieldValueType>;
 /* ****************************** *
  * Field Type Restriction Objects *
  * ****************************** */
-export const StringFieldRestrictions = zod
+export const BooleanFieldRestrictions = zod
+	.object({ empty: zod.boolean(), required: zod.boolean() })
+	.partial()
+	.strict();
+export type BooleanFieldRestrictions = zod.infer<typeof BooleanFieldRestrictions>;
+
+const BooleanFieldConditionalRestriction: ZodType<ConditionalRestriction<BooleanFieldRestrictions>> = zod
 	.object({
-		codeList: RestrictionCodeListString.or(ReferenceTag),
+		if: ConditionalRestrictionTest,
+		then: BooleanFieldRestrictions.or(zod.lazy(() => BooleanFieldConditionalRestriction))
+			.or(zod.array(zod.union([BooleanFieldRestrictions, zod.lazy(() => BooleanFieldConditionalRestriction)])))
+			.optional(),
+		else: BooleanFieldRestrictions.or(zod.lazy(() => BooleanFieldConditionalRestriction))
+			.or(zod.array(zod.union([BooleanFieldRestrictions, zod.lazy(() => BooleanFieldConditionalRestriction)])))
+			.optional(),
+	})
+	.strict();
+
+const BooleanFieldRestrictionsObject = BooleanFieldRestrictions.or(BooleanFieldConditionalRestriction);
+export type BooleanFieldRestrictionsObject = zod.infer<typeof BooleanFieldRestrictionsObject>;
+
+export const IntegerFieldRestrictions = zod
+	.object({
+		codeList: RestrictionCodeListInteger.or(ReferenceTag),
+		empty: zod.boolean(),
 		required: zod.boolean(),
-		regex: RestrictionRegex.or(ReferenceTag),
+		range: RestrictionIntegerRange,
 	})
-	.partial();
-export type StringFieldRestrictions = zod.infer<typeof StringFieldRestrictions>;
+	.partial()
+	.strict();
+export type IntegerFieldRestrictions = zod.infer<typeof IntegerFieldRestrictions>;
+
+const IntegerFieldConditionalRestriction: ZodType<ConditionalRestriction<IntegerFieldRestrictions>> = zod
+	.object({
+		if: ConditionalRestrictionTest,
+		then: IntegerFieldRestrictions.or(zod.lazy(() => IntegerFieldConditionalRestriction))
+			.or(zod.array(zod.union([IntegerFieldRestrictions, zod.lazy(() => IntegerFieldConditionalRestriction)])))
+			.optional(),
+		else: IntegerFieldRestrictions.or(zod.lazy(() => IntegerFieldConditionalRestriction))
+			.or(zod.array(zod.union([IntegerFieldRestrictions, zod.lazy(() => IntegerFieldConditionalRestriction)])))
+			.optional(),
+	})
+	.strict();
```

Comment:
This is how the conditional restrictions are defined. They use `zod.lazy()` to reference themselves since this is a recursive type definition... you can have a conditional restriction inside the `then` or `else` property of the condition. A similar object is defined for each field type (string/number/integer/boolean).